lock/redis/standalone.go

```diff
@@ -135,37 +133,35 @@ func (r *StandaloneRedisLock) parseConnectedSlaves(res string) int {
 // Try to acquire a redis lock.
 func (r *StandaloneRedisLock) TryLock(req *lock.TryLockRequest) (*lock.TryLockResponse, error) {
 	// 1.Setting redis expiration time
-	nx := r.client.SetNX(r.ctx, req.ResourceID, req.LockOwner, time.Second*time.Duration(req.ExpiryInSeconds))
-	if nx == nil {
+	nxval, err := r.client.SetNX(r.ctx, req.ResourceID, req.LockOwner, time.Second*time.Duration(req.ExpiryInSeconds))
+	if nxval == nil {
 		return &lock.TryLockResponse{}, fmt.Errorf("[standaloneRedisLock]: SetNX returned nil.ResourceID: %s", req.ResourceID)
 	}
 	// 2. check error
-	err := nx.Err()
 	if err != nil {
 		return &lock.TryLockResponse{}, err
 	}
 
 	return &lock.TryLockResponse{
-		Success: nx.Val(),
+		Success: *nxval,
 	}, nil
 }
 
 // Try to release a redis lock.
 func (r *StandaloneRedisLock) Unlock(req *lock.UnlockRequest) (*lock.UnlockResponse, error) {
 	// 1. delegate to client.eval lua script
-	eval := r.client.Eval(r.ctx, unlockScript, []string{req.ResourceID}, req.LockOwner)
+	evalInt, parseErr, err := r.client.EvalInt(r.ctx, unlockScript, []string{req.ResourceID}, req.LockOwner)
 	// 2. check error
-	if eval == nil {
+	if evalInt == nil {
 		return newInternalErrorUnlockResponse(), fmt.Errorf("[standaloneRedisLock]: Eval unlock script returned nil.ResourceID: %s", req.ResourceID)
 	}
-	err := eval.Err()
 	if err != nil {
 		return newInternalErrorUnlockResponse(), err
 	}
 	// 3. parse result
-	i, err := eval.Int()
+	i := *evalInt
 	status := lock.InternalError
-	if err != nil {
+	if parseErr != nil {
 		return &lock.UnlockResponse{
 			Status: status,
 		}, err
```

Comment:
Can't leave a comment, but on L 193 (`c.client.Close()`) consider setting c.client to nil. Same on L189 after cancling the context

Reply:
193 makes sense, but I think for 189 we should still let it attempt to close the client first.